src/main/java/com/amazonaws/AmazonECRClientFactory.java

```diff
@@ -0,0 +1,54 @@
+package com.amazonaws;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.services.ecr.AmazonECRClient;
+import hudson.ProxyConfiguration;
+import jenkins.model.Jenkins;
+
+/**
+ * Factory to use Jenkins Proxy configuration with a AmazonECRClient.
+ *
+ * @author Marcus Wallin, kuisathaverat
+ */
+public class AmazonECRClientFactory {
+    @edu.umd.cs.findbugs.annotations.SuppressFBWarnings(
+            value="NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE",
+            justification="The null value it is checked")
+    private final static class ClientConfigurationWrapper extends ClientConfiguration {
```

Comment:
You don't need a subclass. Just use a factory method to configure the ClientConfiguration object.